src/stripe/methods/core/webhook/webhook.cr

```diff
@@ -1,112 +1,106 @@
+                           tolerance : Int32? = DEFAULT_TOLERANCE)
+    Signature.verify_header(payload: payload, header: sig_header, secret: secret, tolerance: tolerance)
 
-    module Signature
-      EXPECTED_SCHEME = "v1"
+    Stripe::Event.from_json(payload)
+  end
 
-      # Computes a webhook signature given a time (probably the current time),
-      # a payload, and a signing secret.
-      def self.compute_signature(timestamp, payload, secret)
-        raise ArgumentError, "timestamp should be an instance of Time" \
-          unless timestamp.is_a?(Time)
-        raise ArgumentError, "payload should be a string" \
-          unless payload.is_a?(String)
-        raise ArgumentError, "secret should be a string" \
-          unless secret.is_a?(String)
+  module Signature
+    EXPECTED_SCHEME = "v1"
 
-        timestamped_payload = "#{timestamp.to_i}.#{payload}"
-        OpenSSL::HMAC.hexdigest(OpenSSL::Digest.new("sha256"), secret,
-                                timestamped_payload)
-      end
+    # Computes a webhook signature given a time (probably the current time),
+    # a payload, and a signing secret.
+    def self.compute_signature(timestamp : Time, payload : String, secret : String) : String
+      timestamped_payload = "#{timestamp.to_unix.to_i}.#{payload}"
+      OpenSSL::HMAC.hexdigest(OpenSSL::Algorithm.new(:SHA256), secret,
+        timestamped_payload)
+    end
 
-      # Generates a value that would be added to a `Stripe-Signature` for a
-      # given webhook payload.
-      #
-      # Note that this isn't needed to verify webhooks in any way, and is
-      # mainly here for use in test cases (those that are both within this
-      # project and without).
-      def self.generate_header(timestamp, signature, scheme = EXPECTED_SCHEME)
-        raise ArgumentError, "timestamp should be an instance of Time" \
-          unless timestamp.is_a?(Time)
-        raise ArgumentError, "signature should be a string" \
-          unless signature.is_a?(String)
-        raise ArgumentError, "scheme should be a string" \
-          unless scheme.is_a?(String)
+    # Generates a value that would be added to a `Stripe-Signature` for a
+    # given webhook payload.
+    #
+    # Note that this isn't needed to verify webhooks in any way, and is
+    # mainly here for use in test cases (those that are both within this
+    # project and without).
+    def self.generate_header(timestamp : Time, signature : String, scheme : String = EXPECTED_SCHEME) : String
+      "t=#{timestamp.to_unix},#{scheme}=#{signature}"
+    end
 
-        "t=#{timestamp.to_i},#{scheme}=#{signature}"
-      end
+    # Extracts the timestamp and the signature(s) with the desired scheme
+    # from the header
+    private def self.get_timestamp_and_signatures(header : String, scheme : String) : Tuple(Time, Array(String))
+      list_items = header.split(',').map { |i| i.split("=", 2) } # ameba:disable Style/VerboseBlock
```

Comment:
I think that if you're just using a generic `i` for the block variable anyways, let's keep to the Ameba standards.

```suggestion
      list_items = header.split(',').map(&.split("=", 2))
```

If you think it's more readable the other way, maybe we could extract and name a method to make the intent more clear?

Reply:
Thanks for the help, I couldn't find the correct short syntax on my own.